src/LessMsi.Cli/Program.cs

```diff
@@ -95,25 +101,81 @@ public static int Main(string[] args)
-		/// </summary>
-		/// <param name="msiFileName">The path of the specified MSI file.</param>
-		/// <param name="outDirName">The directory to extract to. If empty it will use the current directory.</param>
-		/// <param name="filesToExtract">The files to be extracted from the msi. If empty all files will be extracted.</param>
-		public static void DoExtraction(string msiFileName, string outDirName, List<string> filesToExtract )
-		{
-			if (string.IsNullOrEmpty(outDirName))
-				outDirName = Path.GetFileNameWithoutExtension(msiFileName);
-			EnsureFileRooted(ref msiFileName);
-			EnsureFileRooted(ref outDirName);
+        /// <summary>
+        /// Extracts all files contained in the specified .msi file into the specified output directory.
+        /// </summary>
+        /// <param name="msiFileName">The path of the specified MSI file.</param>
+        /// <param name="outDirName">The directory to extract to. If empty it will use the current directory.</param>
+        /// <param name="filesToExtract">The files to be extracted from the msi. If empty all files will be extracted.</param>
+        /// /// <param name="extractionMode">Enum value for files extraction without folder structure</param>
+        public static void DoExtraction(string msiFileName, string outDirName, List<string> filesToExtract, eExtractionMode extractionMode)
+        {
+            if (string.IsNullOrEmpty(outDirName))
+                outDirName = Path.GetFileNameWithoutExtension(msiFileName);
+
+            EnsureFileRooted(ref msiFileName);
+            EnsureFileRooted(ref outDirName, extractionMode);
 
             var msiFile = new LessIO.Path(msiFileName);
 
-			Console.WriteLine("Extracting \'" + msiFile + "\' to \'" + outDirName + "\'.");
+            Console.WriteLine("Extracting \'" + msiFile + "\' to \'" + outDirName + "\'.");
 
-			Wixtracts.ExtractFiles(msiFile, outDirName, filesToExtract.ToArray(), PrintProgress);
-		}
+            Wixtracts.ExtractFiles(msiFile, outDirName, filesToExtract.ToArray(), PrintProgress);
+
+            if (isExtractionModeFlat(extractionMode))
+            {
+                Dictionary<string, int> fileNameCountingDict = new Dictionary<string, int>();
+
+                string originalOutDirName = outDirName.Remove(outDirName.Length - c_TempFolderSuffix.Length) + "\\";
+                Directory.CreateDirectory(originalOutDirName);
+
+                copyFilesInFlatWay(outDirName, originalOutDirName, extractionMode, fileNameCountingDict);
+
+                Directory.Delete(outDirName, true);
+            }
+        }
+
+        private static bool isExtractionModeFlat(eExtractionMode extractionMode)
+        {
+            bool flatExtractionModeFlag = extractionMode == eExtractionMode.OverwriteFlatExtraction;
+            flatExtractionModeFlag |= extractionMode == eExtractionMode.RenameFlatExtraction;
+
+            return flatExtractionModeFlag;

```

Comment:
return eExtractionMode.OverwriteFlatExtraction || eExtractionMode.RenameFlatExtraction;

Reply:
Will amend the code accordingly.

Reply:
`return eExtractionMode.OverwriteFlatExtraction || eExtractionMode.RenameFlatExtraction;` 
does not make sense as the enum values `OverwriteFlatExtraction` and `RenameFlatExtraction` are not Boolean and does not check the value of the passed parameter `extractionMode`.

Can I instead offer the next expression:
`return extractionMode == ExtractionMode.RenameFlatExtraction || extractionMode == ExtractionMode.OverwriteFlatExtraction`

This expression compares the parameter `extractionMode` to the relevant `OverwriteFlatExtraction` and `RenameFlatExtraction` values.